### ISSUE TITLE
After restart i lost all my data

### ISSUE THREAD

**Issue by wingser** [MUnique/OpenMU#386]
**Describe the bug**
Today , i restart my openmu server ,after that ,i found all my config is gone, all the data were init again .

**To Reproduce**

before i restart my openmu server , i set the debian's system lange, idont know whether thats the problem.
and ,i visit the mu panel by a new port(before i visit by the ngnix port , today i found the 81 port of munique/openmu can visit the panel and i tried)

i dont konw wether the problem can be reproduced ,but this is realy a serious problem

**Expected behavior**

**Screenshots**
my account is missing . all data was init again.
and the store i modified before was also init again.
![image](https://github.com/MUnique/OpenMU/assets/13481389/70d48d10-e0be-4acd-85f5-448b5e118367)
![image](https://github.com/MUnique/OpenMU/assets/13481389/a1b826a9-a52d-488e-8b51-17a1eeb1dc63)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] debian 
 - Browser [e.g. chrome, safari] chrome

**Smartphone (please complete the following information):**

**Additional context**

**Comment by lucasfcnunes** [comment]
TL;DR
On docker, be sure to volume mount the correct path for the postgres database persistance.

---
Did you use docker (containers)? If so, you need to set up a volume to persist data across restarts. Storage/data inside containers is ephemeral by default.

https://docs.docker.com/storage

> - The data doesn't persist when that container no longer exists, and it can be difficult to get the data out of the container if another process needs it.

> Docker has two options for containers to store files on the host machine, so that the files are persisted even after the container stops: volumes, and bind mounts.

```compose.yaml
volumes:
  dbdata:

services:
  # ...
  database:
    image: postgres
    container_name: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: admin
      POSTGRES_DB: openmu
      POSTGRES_USER: postgres
    ports:
      - "5432"
    volumes:
      - dbdata:/var/lib/postgresql/data
  # ...
```

**Comment by wingser** [comment]
Thank u，I know the problem now. but i think it would be better config in docker compose file berfore .  then it will never happen to any one else.



***@***.***
 
From: Lucas Fernando Cardoso Nunes
Date: 2023-11-06 19:29
To: MUnique/OpenMU
CC: wingser; Author
Subject: Re: [MUnique/OpenMU] After restart i lost all my data (Issue #386)
TL;DR
On docker, be sure to volume mount the correct path for the postgres database persistance.


Did you use docker (containers)? If so, you need to set up a volume to persist data across restarts. Storage/data inside containers is ephemeral by default.
https://docs.docker.com/storage
The data doesn't persist when that container no longer exists, and it can be difficult to get the data out of the container if another process needs it.
Docker has two options for containers to store files on the host machine, so that the files are persisted even after the container stops: volumes, and bind mounts.
volumes:
  dbdata:
services:
  # ...
  database:
    image: postgres
    container_name: postgres
    restart: always
    environment:
      POSTGRES_PASSWORD: admin
      POSTGRES_DB: openmu
      POSTGRES_USER: postgres
    ports:
      - "5432"
    volumes:
      - dbdata:/var/lib/postgresql/data
  # ...

**Comment by sven-n** [comment]
Thanks @lucasfcnunes for this explaination.
When using the provided docker-compose files, the data of the postgres container is persisted in volumes. I assume you deleted them, or you setup the server in demo mode.
